Providers/Scripts/2.6x-2.7x/Scripts/nxOMSAgentNPMConfig.py

```diff
@@ -0,0 +1,215 @@
+    return retval
+
+
+def Get_Marshall(ConfigType, ConfigID, Contents, Ensure, ContentChecksum):
+    arg_names = list(locals().keys())
+    (ConfigType, ConfigID, Contents, Ensure, ContentChecksum) = init_vars(ConfigType, ConfigID, Contents, Ensure, ContentChecksum)
+    retval = 0
+    retval = Get(ConfigType, ConfigID, Contents, Ensure, ContentChecksum)
+    ConfigType = protocol.MI_String(ConfigType)
+    ConfigID = protocol.MI_String(ConfigID)
+    Ensure = protocol.MI_String(Ensure)
+    Contents = protocol.MI_String(Contents)
+    ContentChecksum = protocol.MI_String(ContentChecksum)
+
+    retd = {}
+    ld = locals()
+    for k in arg_names:
+        retd[k] = ld[k]
+    return retval, retd
+
+
+############################################################
+# Begin user defined DSC functions
+############################################################
+
+def SetShowMof(a):
+    global show_mof
+    show_mof = a
+
+
+def ShowMof(op, ConfigType, ConfigID, Contents, Ensure, ContentChecksum):
+    if not show_mof:
+        return
+    mof = ''
+    mof += op + ' nxOMSAgentNPMConfig MyNPMConfig \n'
+    mof += '{\n'
+    mof += '    ConfigType = "' + ConfigType + '"\n'
+    mof += '    ConfigID = "' + ConfigID + '"\n'
+    mof += '    Contents = "' + Contents + '"\n'
+    mof += '    Ensure = "' + Ensure + '"\n'
+    mof += '    ContentChecksum = "' + ContentChecksum + '"\n'
+    mof += '}\n'
+    f = open('./test_mofs.log', 'a')
+    Print(mof, file=f)
+    LG().Log('INFO', mof)
+    f.close()
+
+
+def Set(ConfigType, ConfigID, Contents, Ensure, ContentChecksum):
+    ShowMof('SET', ConfigType, ConfigID, Contents, Ensure, ContentChecksum)
+    retval = 0
+    destFileFullPath = CONFIG_PATH.__add__(DEST_FILE_NAME)
+    if ConfigType != 'UpdateNPMAgentConfig':
+        LG().Log('ERROR', 'Config type did not match, exiting set')
+        return [retval]
+    if Ensure != 'Present':
+        LG().Log('INFO', 'Ensure is not present, exiting set')
+        return [retval]
+
+    # create directory 
+    if not os.path.exists(CONFIG_PATH):
+        LG().Log('ERROR', 'Path does not exist creating directory')
+        #TODO: mkdir to be done via sudo shellscript
+        os.mkdir(CONFIG_PATH)
+    retval = WriteFile(destFileFullPath, Contents)
+    if retval == 0:
+        LG().Log('INFO', 'Updated the file, going to notify server')
+        retval = NotifyServer()
+    return [retval]
+
+
+def Test(ConfigType, ConfigID, Contents, Ensure, ContentChecksum):
+    ShowMof('TEST', ConfigType, ConfigID, Contents, Ensure, ContentChecksum)
+    retval = 0
+    destFileFullPath = CONFIG_PATH.__add__(DEST_FILE_NAME)
+    if ConfigType != 'UpdateNPMAgentConfig':
+        LG().Log('ERROR', 'Config type did not match, exiting test')
+        return [retval]
+    if Ensure != 'Present':
+        LG().Log('INFO', 'Ensure is not present, exiting test')
+        return [retval]
+
+    if not os.path.exists(destFileFullPath):
```

Comment:
destFileFullPath checks whether config file exists or not. This would be a day 0 scenario where config needs to be pushed for the first time. I have added a check to CONFIG_PATH separately.